spring-cql/src/test/java/org/springframework/cassandra/config/PoolingOptionsFactoryBeanUnitTests.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cassandra.config;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyInt;
+import static org.mockito.Mockito.*;
+
+import java.util.concurrent.Executor;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.datastax.driver.core.HostDistance;
+import com.datastax.driver.core.PoolingOptions;
+
+/**
+ * Unit tests for {@link PoolingOptionsFactoryBean}.
+ *
+ * @author Sumit Kumar
+ * @author David Webb
+ * @author John Blum
+ * @see org.springframework.cassandra.config.PoolingOptionsFactoryBean
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class PoolingOptionsFactoryBeanUnitTests {
+
+	@Mock
+	private Executor mockExecutor;
+
+	@Mock
+	private PoolingOptions mockPoolingOptions;
+
+	private PoolingOptionsFactoryBean poolingOptionsFactoryBean;
+
+	@Before
+	public void setup() {
+		poolingOptionsFactoryBean = new PoolingOptionsFactoryBean();
+	}
+
+	@Test
+	public void isSingletonIsTrue() {
+		assertThat(poolingOptionsFactoryBean.isSingleton(), is(true));
+	}
+
+	@Test
+	public void getObjectReturnsNullWhenNotInitialized() throws Exception {
+		assertThat(poolingOptionsFactoryBean.getObject(), is(nullValue(PoolingOptions.class)));
+	}
+
+	@Test
```

Comment:
The other test case methods: `getObjectReturnsNullWhenNotInitialized`, `getObjectTypeReturnsPoolingOptionsClassWhenNotInitialized` and `isSingletonIsTrue` have nothing to do with [DATACASS-298](https://jira.spring.io/browse/DATACASS-298).  They were added because the overall `PoolingOptionsFactoryBean` class was missing (and still is to a certain extent) adequate unit test coverage.